### PR TITLE
Maya: Deadline custom settings 

### PR DESCRIPTION
**Pull request by antirotor**
This PR adds ability to customize Deadlines `Job_Info` and `Plugin_Info` from presets.

It will update existing data prior to submission, useful for adding / overriding

◀️ **Pype 2 backport**: #1655
|---|

### CODE REVIEW

File: openpype/modules/deadline/plugins/publish/submit_maya_deadline.py [hunk 1]
```diff
@@ -271,6 +271,21 @@ def process(self, instance):
             ["DEADLINE_REST_URL"]
         )
 
+        self._job_info = (
+            context.data["project_settings"]
+            ["maya"]
+            ["publish"]
+            ["deadline"]
+            ["jobInfo"]
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/plugins/publish/submit_maya_deadline.py:279]: I would add `.get("jobInfo", {})` for cases that settings are not updated yet (Same for `"pluginInfo"`).

File: openpype/settings/entities/schemas/projects_schema/schemas/schema_maya_publish.json [hunk 2]
```diff
@@ -309,6 +309,24 @@
             "type": "schema",
             "name": "schema_maya_capture"
         },
+        {
```
Comment (iLLiCiTiT) [inline openpype/settings/entities/schemas/projects_schema/schemas/schema_maya_publish.json:312]: There is `Submit maya job to deadline` in `Deadline` settings to separate deadline.

image

**Comment by iLLiCiTiT** [comment]
Question not related to this PR but do we know why **Group name** is set to string `"none"` and not empty string by default?

**Comment by antirotor** [comment]
> Question not related to this PR but do we know why **Group name** is set to string `"none"` and not empty string by default?

that is how Deadline handles it as default value :) see:
https://docs.thinkboxsoftware.com/products/deadline/10.0/1_User%20Manual/manual/manual-submission.html#general

**Comment by iLLiCiTiT** [comment]
> that is how Deadline handles it as default value :) see:

I see but in the code is:
```
if self.group != "none":
    self.payload_skeleton["JobInfo"]["Group"] = self.group
```
So `"Group"` is not set only if `group` is `"none"`. I would just change the condition to `if self.group and self.group != "none":` to make it valid when it's empty string.

**Review by iLLiCiTiT — APPROVED** [review]
Approved

**Pull request merged**